src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -323,6 +338,83 @@ private static IEnumerable<VisualStudioInstance> GetInstances(VisualStudioInstan
 #endif
         }
 
+        static IEnumerable<VisualStudioInstance> GetMonoMSBuildInstances ()
+        {
+            // $prefix/lib/mono/4.5/mscorlib.dll
+            var runningMonoPath = Path.GetFullPath(Path.Combine(Path.GetDirectoryName(typeof (object).Assembly.Location), "..", "..", ".."));
+            if (TryGetValidMonoVersion (runningMonoPath, out var version))
+            {
+                yield return new VisualStudioInstance("Mono", runningMonoPath, version, DiscoveryType.Mono);
+            }
+
+            if (!IsOSX)
+            {
+                // Returning just one instance on !osx
+                yield break;
+            }
+
+            foreach(var dirPath in Directory.EnumerateDirectories(s_monoOSXBasePath))
+            {
+                if (string.Compare(Path.GetFileName (dirPath), "Current") == 0 || // skip the `Current` symlink
+                    string.Compare(dirPath, runningMonoPath) == 0)               // and the running mono version
```

Comment:
nit:
```suggestion
                if (string.Equals(Path.GetFileName(dirPath), "Current") || // skip the `Current` symlink
                    string.Equals(dirPath, runningMonoPath))                   // and the running mono version
```

Also, this feels a little fragile because it would fail if someone appends a slash to dirPath. That may sound arbitrary, but other places we make a new VisualStudioInstance append a slash to the end in setting the path, and people can access that directly using QueryVisualStudioInstances, so not having that would be a confusing difference in how it works.

Reply:
Append a `/` to the directory name itself? `dirPath` is from `Directory.EnumerateDirectories`

Reply:
Right. The two scenarios I'm talking about here are:
1) Someone expects dirPath to end in a slash because it does for other runtimes. They have code that uses that --> error.
2) Someone notes that dirPath ends in a slash for other runtimes and decides to centralize it with some method that's called before this. Slash is added to dirPath --> Path.GetFileName returns an empty string --> this part fails.

The first is a current problem with this code, and the second is a hypothetical I'd like to avoid. Is that clearer?

Reply:
Ok. Changed it to `string.Equals(Path.GetFullPath(dirPath), runningMonoFullPath)`